tests/chkstk-test.c

```diff
@@ -2,12 +2,17 @@
    Tests stack probing
 */
 
-void f1(char* data) 
+#define BIG_STACK_ALLOC_SIZE 300000
+
+void write_to_big_stack_alloc(char* stack_data) 
 {
-    *data = 0;
+    stack_data[0] = 18;
+    stack_data[BIG_STACK_ALLOC_SIZE - 1] = 19;
 }
 
-void chkstk_test() {
-    char test[300000];
-    f1(test);
+int check_big_stack_allocation() 
+{
+    char stack_data[BIG_STACK_ALLOC_SIZE];
+    write_to_big_stack_alloc(stack_data);
+    return stack_data[BIG_STACK_ALLOC_SIZE - 1] == 19;

```

Comment:
Also this check does not make sense as it is checking value 19 on the top of the stack which is not effected in this case.

Reply:
As you can see the function write_to_big_stack_alloc writes the value 19 to this location in stack_data. Maybe I am confused. Could you expand by what you mean here?

Reply:
sorry for confusion 
stack_data[BIG_STACK_ALLOC_SIZE - 1] == 19 checks bottom of the stack which is not affected by missing data probing issue

Reply:
This PR is getting large. Will follow-up in next PR to avoid merging issues.